internal/security/kdf/interface.go

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package kdf
+
+// KeyDeriver is the interface that the main program expects
+// for returning a derived key
+type KeyDeriver interface {
+	// DeriveKey returns a byte array that is of keyLen length and
+	// an error if errors where encountered while deriving the key
+	DeriveKey(ikm []byte, keylkeyLen uint, info string) ([]byte, error)
```

Comment:
> Is the 2nd parameter supposed to be `keyLen` (v.s. `keylkeyLen`)?

Typo fixed.

> Also why type of key is being returned, symmetric, public, private? Do algorithms matter here?

The key is algorithm-less -- it is a bunch of random stuff that could be used as a symmetric key, a password, input to a second KDF, et cetera.  It is for sure not a public/private keypair, however.

> It would also be helpful to document the parameters of DeriveKey

Added

Reply:
> The commit itself is a single commit summary, at least include a "fixes: 1919" in the body of your commit message, although an actual commit message body with a little more context would be nice.
> I assume you intend to squash the "fixup!" commits before landing?

Both of these are addressed.